### PR TITLE
os/board/rtl8720e: Update BT to AP+NP architecture

### PR DESCRIPTION
**Pull request by Dracaryser**
Update BT to AP+NP architecture, AP part on tizenrt and NP part on freertos(NP part had been updated by gexuyan)

Signed-off-by: roy_zhu <roy_zhu@realsil.com.cn>

### CODE REVIEW

File: build/configs/rtl8720e/hello/defconfig [hunk 1]
```diff
@@ -142,7 +142,7 @@ CONFIG_INIC_IPC_HIGH_TP=1
 # Realtek RTL8720E BLE Support
 #
 CONFIG_AMEBALITE_BLE=y
-CONFIG_FTL_ENABLED=y
+# CONFIG_FTL_ENABLED=y
```
Comment (sunghan-chang) [inline build/configs/rtl8720e/hello/defconfig:145]: Please don't update this manually. After updating .config through the menuconfig, please copy .config into defconfig.

File: os/board/rtl8720e/src/component/bluetooth/example/utils/bt_utils.c [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+		byte_low = ctoi(*(str_ptr++));
+		if (byte_high == 0xFF || byte_low == 0xFF) {
+			return FALSE;
+		}
+		byte_arr[n/2] = byte_high << 4 | byte_low;
+		n+=2;
+	}
+
+	return TRUE;
+}
```
Comment (sunghan-chang) [inline os/board/rtl8720e/src/component/bluetooth/example/utils/bt_utils.c:139]: Could you check `no newline at end of file` warning?

File: os/board/rtl8720e/src/rtl8720e_boot.c [hunk 3]
```diff
@@ -336,6 +336,9 @@ void board_initialize(void)
 #ifdef CONFIG_AMEBALITE_WIFI
 	wlan_initialize();
 #endif
+#if defined(CONFIG_BT_AP) && CONFIG_BT_AP
```
Comment (sunghan-chang) [inline os/board/rtl8720e/src/rtl8720e_boot.c:339]: Could you check whether proper header (it looks `os/board/rtl8720e/src/project/realtek_amebaLite_va0_example/inc/inc_km4/platform_autoconf.h`) is included well or not? I can't find that in header including.

**Comment by sunghan-chang** [comment]
@Dracaryser 
1. Could you check a build break on the loadable_apps?
2. Could you let us know BLE functionality test result?
3. How is memory usages in flash and in ram?

**Comment by sunghan-chang** [comment]
> (NP part had been updated by gexuyan)

@Dracaryser Could you let me know what PR was? Did the PR say that in the description??

**Comment by gexuyan** [comment]
> 

Dear @sunghan-chang ,
The NP part Roy mentioned is in PR 5710 and has been merged into master branch already. This PR mentioned KR4 image has enabled BT configs in the description. Thank you.

**Comment by Dracaryser** [comment]
Dear @sunghan-chang ,
1.Could you check a build break on the loadable_apps?
=>I have fixed the build error and the warning you mentioned above.
2.Could you let us know BLE functionality test result?
=>I have tested each API in ble tizenrt_server.c, ble_tizenrt_client.c and ble_tizenrt_combo.c to make sure every basic function works well.
3.How is memory usages in flash and in ram?
=> @yeetee179 is helping to check the memory usage in flash and ram，he will show the result when it's done.
Thanks.

File: os/board/rtl8720e/src/rtl8720e_boot.c (hunk 3 above)
Comment (Dracaryser) [reply os/board/rtl8720e/src/rtl8720e_boot.c]: I have replaced CONFIG_BT_AP with CONFIG_AMEBALITE_BLE which is defined in config.h.

File: build/configs/rtl8720e/hello/defconfig (hunk 1 above)
Comment (Dracaryser) [reply build/configs/rtl8720e/hello/defconfig]: Thanks for your help, I have fixed it as you said.

**Comment by yeetee179** [comment]
Dear @sunghan-chang ,
The following is the flash and ram usage:
hello example flash size: 1616736 bytes
hello example ram size: 92288 bytes

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**